test/test_blocks.py

```diff
@@ -86,7 +87,11 @@ def fake_func():
 
         config = demo.get_config_file()
         config.pop("version")  # remove version key
```

Comment:
hmm, instead of popping `version` in individual tests, wouldn't it be better to do it in the `assert_configs_are_equivalent_besides_ids`?

Reply:
Yea I think it will be better if the logic for ignoring fields is in `assert_configs_are_equivalent..` as opposed to the calling code. Maybe we can add a `ignore_fields: List[str]` parameter to the function that gives uses the ability to specify what they want to ignore. I can see use cases where you would want to ignore the version key and others where you wouldn't. 

I don't think this is necessary to merge this PR though. If anything I can file an issue to implement in the future.

Reply:
Sounds good! Removed the line and added a parameter `root_keys` which specifies which keys to test for equivlance